snowflake_test.go

```diff
@@ -0,0 +1,368 @@
+
+	password := "y8fva_sdVA3rasf"
+
+	createReq := dbplugin.NewUserRequest{
+		UsernameConfig: dbplugin.UsernameMetadata{
+			DisplayName: "test",
+			RoleName:    "test",
+		},
+		Statements: dbplugin.Statements{
+			Commands: []string{`
+				CREATE USER {{name}} PASSWORD = '{{password}}';
+				GRANT ROLE myrole TO USER {{name}};`,
+			},
+		},
+		Password:   password,
+		Expiration: time.Now().Add(time.Hour),
+	}
+
+	createResp := dbtesting.AssertNewUser(t, db, createReq)
+
+	assertCredentialsExist(t, connURL, createResp.Username, password)
+
+	renewReq := dbplugin.UpdateUserRequest{
+		Username: createResp.Username,
+		Expiration: &dbplugin.ChangeExpiration{
+			NewExpiration: time.Now().Add(time.Minute),
+		},
+	}
+
+	dbtesting.AssertUpdateUser(t, db, renewReq)
+
+	// Sleep longer than the initial expiration time
+	time.Sleep(2 * time.Second)
+
+	assertCredentialsExist(t, connURL, createResp.Username, password)
+	attemptDropUser(connURL, createResp.Username)
+}
+
+func TestSnowflake_RevokeUser(t *testing.T) {
+	if !runAcceptanceTests {
+		t.SkipNow()
+	}
+
+	connURL := connUrl(t)
+
+	type testCase struct {
+		deleteStatements []string
+	}
+
+	tests := map[string]testCase{
+		"name revoke": {
+			deleteStatements: []string{`
+				DROP USER {{name}};`,
+			},
+		},
+		"username revoke": {
+			deleteStatements: []string{`
+				DROP USER {{username}};`,
+			},
+		},
+		"default revoke": {},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			db := new()
+			defer dbtesting.AssertClose(t, db)
+
+			initReq := dbplugin.InitializeRequest{
+				Config: map[string]interface{}{
+					"connection_url": connURL,
+				},
+				VerifyConnection: true,
+			}
+			dbtesting.AssertInitialize(t, db, initReq)
+
+			password := "y8fva_sdVA3rasf"
+
+			createReq := dbplugin.NewUserRequest{
+				UsernameConfig: dbplugin.UsernameMetadata{
+					DisplayName: "test",
+					RoleName:    "test",
+				},
+				Statements: dbplugin.Statements{
+					Commands: []string{`
+						CREATE USER {{name}} PASSWORD = '{{password}}';
+						GRANT ROLE myrole TO USER {{name}};`,
+					},
+				},
+				Password:   password,
+				Expiration: time.Now().Add(time.Hour),
+			}
+
+			createResp := dbtesting.AssertNewUser(t, db, createReq)
+
+			assertCredentialsExist(t, connURL, createResp.Username, password)
+
+			deleteReq := dbplugin.DeleteUserRequest{
+				Username: createResp.Username,
+				Statements: dbplugin.Statements{
+					Commands: test.deleteStatements,
+				},
+			}
+			dbtesting.AssertDeleteUser(t, db, deleteReq)
+			assertCredentialsDoNotExist(t, connURL, createResp.Username, password)
+		})
+	}
+}
+
+func dsnString() (string, error) {
+	user := os.Getenv(envVarSnowflakeUser)
+	password := os.Getenv(envVarSnowflakePassword)
+	account := os.Getenv(envVarSnowflakeAccount)
+
+	if user == "" || password == "" || account == "" {
+		err := errors.New("critical env vars were not set. user, password, and account must all be set")
```

Comment:
nit: It may be helpful to log the actual missing env var names (e.g., `SNOWFLAKE_USER`, `SNOWFLAKE_PASSWORD`, etc) that need to be set instead of the variable names used here.

Thanks for putting the env vars in the README too, btw 👍